### ISSUE TITLE
[YSQL] Insert into partitioned table might succeed even on constraint failure

### ISSUE THREAD

**Issue by frozenspider** [yugabyte/yugabyte-db#5387]
Row-level partition support was implemented in #5179

Consider this (excerpt from `foreign_key` test):
```sql
CREATE TABLE fk_notpartitioned_pk (a int, b int, primary key (a, b));
CREATE TABLE fk_partitioned_fk (a int default 2501, b int default 142857) PARTITION BY LIST (a);
CREATE TABLE fk_partitioned_fk_1 PARTITION OF fk_partitioned_fk FOR VALUES IN (NULL,500,501,502);
ALTER TABLE fk_partitioned_fk ADD FOREIGN KEY (a, b)
  REFERENCES fk_notpartitioned_pk MATCH SIMPLE
  ON DELETE SET NULL ON UPDATE SET NULL;
CREATE TABLE fk_partitioned_fk_2 PARTITION OF fk_partitioned_fk FOR VALUES IN (1500,1502);
CREATE TABLE fk_partitioned_fk_3 (a int, b int);
ALTER TABLE fk_partitioned_fk ATTACH PARTITION fk_partitioned_fk_3 FOR VALUES IN (2500,2501,2502,2503);

INSERT INTO fk_partitioned_fk (a, b) VALUES (2502, 2503);

SELECT tableoid::regclass, a, b FROM fk_partitioned_fk ORDER BY a, b;
```
This INSERT fails:
```
yugabyte=# INSERT INTO fk_partitioned_fk (a, b) VALUES (2502, 2503);
ERROR:  insert or update on table "fk_partitioned_fk_3" violates foreign key constraint "fk_partitioned_fk_a_fkey"
DETAIL:  Key (a, b)=(2502, 2503) is not present in table "fk_notpartitioned_pk".
```
... but SELECT shows that the value was in fact inserted:
```
yugabyte=# SELECT tableoid::regclass, a, b FROM fk_partitioned_fk ORDER BY a, b;
      tableoid       |  a   |  b
---------------------+------+------
 fk_partitioned_fk_3 | 2502 | 2503
(1 row)
```
This diverges from PostgreSQL v11

cc @deeps1991

**Comment by frozenspider** [comment]
Somewhat simpler example:
```sql
DROP TABLE IF EXISTS notpartitioned, partitioned_fk CASCADE;

CREATE TABLE notpartitioned (a int, PRIMARY KEY (a));
CREATE TABLE partitioned_fk (a int) PARTITION BY LIST (a);
ALTER TABLE partitioned_fk ADD FOREIGN KEY (a) REFERENCES notpartitioned MATCH SIMPLE;
CREATE TABLE partitioned_part PARTITION OF partitioned_fk FOR VALUES IN (123);

INSERT INTO partitioned_fk (a) VALUES (123);

SELECT tableoid::regclass, a FROM partitioned_fk ORDER BY a;
```

**Comment by frozenspider** [comment]
Found the cause: this `INSERT` is erroneously classified as a single-row transaction

**Comment by frozenspider** [comment]
Resolved in 5c145a1
